crt_portal/api/filters.py

```diff
@@ -0,0 +1,103 @@
+                            'IER - Form Letter': 0,
+                            'EOS - EEOC Referral Form Letter': 0,
+                            'EOS - Department of Ed OCR Referral Form Letter': 0,
+                            'SPL - Referral for PREA Issues': 0,
+                            'SPL - Standard Form Letter': 0,
+                            'Trending - Arbery Inquiries': 0,
+                            'Trending - Floyd Inquiries': 0,
+                            'Trending - General COVID Inquiries': 0}
+    emails_counter_for_date_range = total_emails_counter.copy()
+
+    contacts_payload = {
+        "start_date": '',
+        "end_date": '',
+        'total_actions': total_actions,
+        'total_contacts': total_contacts,
+        "total_contacts_in_range": 0,
+        "total_actions_in_range": 0,
+        "total_emails_counter": total_emails_counter,
+        "emails_counter_for_date_range": emails_counter_for_date_range
+    }
+
+    for field in querydict.keys():
+        filter_list = querydict.getlist(field)
+        if len(filter_list) > 0:
+            filters[field] = filter_list
+        if 'date' in field:
+            # filters by a start date or an end date expects yyyy-mm-dd
+            field_name = 'timestamp'
+            encoded_date = filter_list[0]
+            contacts_payload[field] = encoded_date
+            decoded_date = urllib.parse.unquote(encoded_date)
+            try:
+                date_obj = datetime.strptime(decoded_date, "%Y-%m-%d")
+                date_obj = _change_datetime_to_end_of_day(date_obj, field)
+                kwargs[f'{field_name}{filter_options[field]}'] = date_obj
+            except ValueError:
+                # if the date is invalid, we ignore it.
+                continue
+    filtered_actions = action_qs.filter(**kwargs).distinct()
+    filtered_contacts = contact_qs.filter(**kwargs).distinct()
+
+    for contact in contact_qs:
+        try:
+            email_title = contact.description.split("'")[1]
+            if email_title:
+                for key in total_emails_counter:
+                    if key == email_title:
+                        total_emails_counter[key] += 1
+        except IndexError:
+            print("oh my, it is not an email")
+
+    for contact in filtered_contacts:
+        try:
+            email_title = contact.description.split("'")[1]
+            if email_title:
+                for key in emails_counter_for_date_range:
+                    if key == email_title:
+                        emails_counter_for_date_range[key] += 1
+        except IndexError:
+            print("oh my, it is not an email")
+
+    contacts_payload['total_contacts_in_range'] = len(filtered_contacts)
+    contacts_payload['total_actions_in_range'] = len(filtered_actions)
+    contacts_payload['total_emails_counter'] = total_emails_counter
+    contacts_payload['emails_counter_for_date_range'] = emails_counter_for_date_range
```

Comment:
I *think* that since I instantiate `emails_counter_for_date_range` on L45, it will exist even if the try fails.  However, maybe I shouldn't create `emails_counter_for_date_range` if there is no date range specified in the path variables?